ML warning with Clang

@trilinos/ml 

```
.../Trilinos/packages/ml/src/Coarsen/ml_agg_min_energy.cpp:48:10: warning: 'register' storage class specifier is deprecated and incompatible with C++1z [-Wdeprecated-register]
  double register dtemp;
         ^~~~~~~~~
```

--- a/packages/ml/src/Coarsen/ml_agg_min_energy.cpp
+++ b/packages/ml/src/Coarsen/ml_agg_min_energy.cpp
@@ -45,7 +45,7 @@ static void ML_multiply_self_all(ML_Operator* Op, double* Column2Norm)
   int nnz     = (data->rowptr)[n_rows];
   int *bindx  = data->columns;
   double *val = data->values;
-  double register dtemp;
+  double dtemp;
 
   for (int i = 0; i < nnz; i++) {
        dtemp = *val++;

